modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqTranslate.scala

```diff
@@ -227,37 +227,40 @@ class SeqTranslate(site: Site, systems: Systems, settings: Settings) {
     else none
   }
 
-  def stopObserve(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd( seqState,
+  def stopObserve(seqId: Sequence.Id)(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd(
+    seqState,
     toInstrumentSys(seqState.toSequence.metadata.instrument).toOption.flatMap(_.observeControl match {
-      case Controllable(StopObserveCmd(stop), _, _, _) => Some(Process.eval(stop.run.map{
+      case Controllable(StopObserveCmd(stop), _, _, _, _, _) => Some(Process.eval(stop.run.map{
         case -\/(e) => Event.logMsg(SeqexecFailure.explain(e))
         case _      => Event.nullEvent
       }))
-      case _                                           => none
+      case _                                                 => none
     } )
-  )
+  ).orElse(stopPaused(seqId)(seqState))
 
-  def abortObserve(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd( seqState,
+  def abortObserve(seqId: Sequence.Id)(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd(
+    seqState,
     toInstrumentSys(seqState.toSequence.metadata.instrument).toOption.flatMap(_.observeControl match {
-      case Controllable(_, AbortObserveCmd(abort), _, _) => Some(Process.eval(abort.run.map{
+      case Controllable(_, AbortObserveCmd(abort), _, _, _, _) => Some(Process.eval(abort.run.map{
         case -\/(e) => Event.logMsg(SeqexecFailure.explain(e))
         case _      => Event.nullEvent
       }))
-      case _                                             => none
+      case _                                                   => none
     } )
-  )
+  ).orElse(abortPaused(seqId)(seqState))
 
   def pauseObserve(seqState: Sequence.State): Option[Process[Task, Event]] = deliverObserveCmd( seqState,
     toInstrumentSys(seqState.toSequence.metadata.instrument).toOption.flatMap(_.observeControl match {
-      case Controllable(_, _, PauseObserveCmd(pause), _) => Some(Process.eval(pause.run.map{
+      case Controllable(_, _, PauseObserveCmd(pause), _, _, _) => Some(Process.eval(pause.run.map{
         case -\/(e) => Event.logMsg(SeqexecFailure.explain(e))
         case _      => Event.nullEvent
       }))
-      case _                                             => none
+      case _                                                   => none
     } )
   )
 
-  def resumeObserve(seqId: Sequence.Id)(seqState: Sequence.State): Option[Process[Task, Event]] = {
+  private def pausedCommand(seqId: Sequence.Id, f: ObserveControl => Option[SeqAction[ObserveCommand.Result]])
```

Comment:
I factored out the code common for resume, stop a paused observation, and abort a paused observation.